test/socketpair_helper.py

```diff
@@ -1,5 +1,13 @@
 import socket
 
+try:
+    _CONNECT_ERROR = (BlockingIOError, InterruptedError)
+except NameError:
+    try:
+        _CONNECT_ERROR = (WinError, OSError, socket.error)  # noqa: F821
+    except NameError:
+        _CONNECT_ERROR = (OSError, socket.error)
```

Comment:
Ok cool, can we get a comment above this block explaining why we're doing this dance? :grin: